### ISSUE TITLE
g:modusline_colors['t'] overrides StatusLineTerm on my colorscheme

### ISSUE THREAD

**Issue by lmintmate** [sunaku/vim-modusline#1]
I installed your plugin, because I wanted to have statusline colors but use a custom statusline, and it worked rather well. However, when I attempted to use the terminal, the color specified by vim-modusline (DiffAdd) overrode the existing color of my terminal statusline, which is defined by StatusLineTerm [in the colorscheme I use](https://github.com/lmintmate/blue-mood-vim).
I attach two images to show how the terminal inside vim looks before and after the installation of the plugin.
Before:
![statuslineterm-before-modusline](https://user-images.githubusercontent.com/26282842/46913778-68820780-cf9c-11e8-99c2-5e3d321407c2.png)
After:
![statuslineterm-after-modusline](https://user-images.githubusercontent.com/26282842/46913780-6f107f00-cf9c-11e8-9f53-dadbaffc8239.png)

If possible, I would like to be able to use the color I set myself, since it fits better with my overall theme (not to mention that DiffAdd is by default also used on insert mode, which can be confusing). _I'm aware that I can just set `let g:modusline_colors['t']      = '' ` to remedy this problem,_ -> edit: scratch that, I can't - when I try this, or the alternative `let g:modusline_colors['t']      = '%#StatusLineTerm#' `, vim basically fails to open.
I believe it would be better if the plugin somehow didn't override the default color if it spotted StatusLineTerm being defined.

System details: gVim 8.1.647 on Linux Mint MATE 18.3 (based on Ubuntu 16.04). 
P.S. The same problem applies for me in terminal vim as well, but only because I use a terminal that supports true color and enable `set termguicolors` in my vimrc, because my colorscheme is more geared towards the gui.

Thanks in advance.

**Comment by lmintmate** [comment]
A clarification: In regards to the problem where I couldn't change `g:modusline_colors['t']`, I experimented some more and found out I needed to put in my vimrc the entirety of the g:modusline_colors variables (as can be seen in the code block in the README) so that vim wouldn't crash. Afterwards, I only changed g:modusline_colors['t'] to `let g:modusline_colors['t'] = '%#StatusLineTerm#'`, which remedies the problem somewhat. It should thus be made clearer in the README that one needs to paste all the variables in the README, even if they only want to change a single one, for this to work. Alternatively, a way to change only a single variable without vim not opening correctly would be highly appreciated. Unfortunately, I don't know enough vimscript myself to attempt anything like that. Thanks again in advance.

**Comment by sunaku** [comment]
Hmm, the singular override *should* work properly if executed *after* this plugin is loaded:
```vim
" Step 1. make sure the plugin is loaded first... 
" (refer to your favorite Vim plugin manager's documentation on how to achieve this)
" Step 2. now you can perform the singular override
let g:modusline_colors['t'] = '%#StatusLineTerm#'
```

Alternatively, you can register a `VimEnter` autocommand hook to run when Vim starts up:
```vim
autocmd VimEnter * call s:customize_modusline()
function! s:customize_modusline() abort
  " do your customization here, inside this function
  let g:modusline_colors['t'] = '%#StatusLineTerm#'
endfunction
```

Alternatively, you can override the `ModuslineColor()` function to differ when modus=`t`:
```vim
function! ModuslineColor(modus) abort
  if a:modus == 't'
    return '%#StatusLineTerm#'
  endif
  return get(g:modusline_colors, a:modus, '%#ErrorMsg#')
endfunction
```

**Comment by lmintmate** [comment]
I use vim plug, and place the plugin settings section (where I put the setting for modusline) of my vimrc after plug#end, so it should normally work - after all, all the other settings there work fine. I used your autocommand solution (didn't try overriding the `ModusLineColor()` function) and this one worked. The autocommand solution seems also better for configuring multiple colors, because in the meanwhile I also changed the look of Replace (`g:modusline_colors['R']`), and simply placed this additional customization inside the function, while with a conditional (which is used in the override method), I'd have to say `if a:modus = 't' or 'R'` or something of that sort and would get it wrong.
tl;dr: I verified that I placed the setting in the correct place in my vimrc. I then tried the autocommand method and it worked.

**Comment by sunaku** [comment]
:+1: I'm glad it worked out, enjoy! :rainbow:
